homeassistant/components/onboarding/views.py

```diff
@@ -60,20 +61,57 @@ async def async_setup(
     hass.http.register_view(BackupInfoView(data))
     hass.http.register_view(RestoreBackupView(data))
     hass.http.register_view(UploadBackupView(data))
-    setup_cloud_views(hass, data)
 
 
-class OnboardingView(HomeAssistantView):
-    """Return the onboarding status."""
+class OnboardingPlatformProtocol(Protocol):
+    """Define the format of onboarding platforms."""
+
+    async def async_setup_views(
+        self, hass: HomeAssistant, data: OnboardingStoreData
+    ) -> None:
+        """Set up onboarding views."""
+
+
+async def async_process_onboarding_platforms(hass: HomeAssistant) -> None:
+    """Start processing onboarding platforms."""
+    await integration_platform.async_process_integration_platforms(
+        hass, DOMAIN, _register_onboarding_platform, wait_for_platforms=True
```

Comment:
I'm not sure we need to set `wait_for_platforms` True here, I think we can skip it.

Reply:
Won't this deadlock if onboarding is stage 0 and need to wait for cloud which is stage 1?

Reply:
I don't think this can deadlock because this is a much softer wait, it's about waiting for platforms for integrations which are already loaded.

Reply:
We don't know that the integrations are loaded when we call this here.

Reply:
Yes, sure.  But it doesn't deadlock.
Anyway, we don't need the wait, I'll remove it.